mantle/kola/tests/rhcos/upgrade.go

```diff
@@ -372,7 +391,7 @@ func downloadLatestReleasedRHCOS(target string) (string, error) {
 
 	var latestOcpRhcosBuild *cosa.Build
 	rhcosVersion := ocpRelease.DisplayVersions.MachineOS.Version
-	latestBaseUrl := fmt.Sprintf("https://rhcos-redirector.apps.art.xq1c.p1.openshiftapps.com/art/storage/releases/rhcos-%s/%s/%s",
+	latestBaseUrl := fmt.Sprintf("https://rhcos.mirror.openshift.com/art/storage/releases/rhcos-%s/%s/%s",
```

Comment:
Right, this is the other broken part of this test. We need to update it for the new layout. But ideally we do it in a way that we also support the old layout until the latest release on any given stream is no longer from there.

Maybe the simplest is to have this code first query the new layout, and then fallback to querying the old one if it gets an error. WDYT?

Anyway, we don't have to handle it here, though it'd be easier to backport if it's in the same PR too.

Reply:
OK, I think basically this:

```diff
diff --git a/mantle/kola/tests/rhcos/upgrade.go b/mantle/kola/tests/rhcos/upgrade.go
index 838a0a7cc..c6b348b8c 100644
--- a/mantle/kola/tests/rhcos/upgrade.go
+++ b/mantle/kola/tests/rhcos/upgrade.go
@@ -391,13 +391,23 @@ func downloadLatestReleasedRHCOS(target string) (string, error) {

        var latestOcpRhcosBuild *cosa.Build
        rhcosVersion := ocpRelease.DisplayVersions.MachineOS.Version
-       latestBaseUrl := fmt.Sprintf("https://rhcos.mirror.openshift.com/art/storage/releases/rhcos-%s/%s/%s",
+       latestBaseUrl := fmt.Sprintf("https://rhcos.mirror.openshift.com/art/storage/prod/streams/%s/builds/%s/%s/meta.json",
                ocpVersionF,
                rhcosVersion,
                coreosarch.CurrentRpmArch())
        latestRhcosBuildMetaUrl := fmt.Sprintf("%s/meta.json", latestBaseUrl)
        if err := getJson(latestRhcosBuildMetaUrl, &latestOcpRhcosBuild); err != nil {
-               return "", err
+               // Try the old bucket layout; ideally we'd only do this if the error
+               // was 403 denied (which is really a 404), but meh this is temporary
+               // anyway.
+               latestBaseUrl = fmt.Sprintf("https://rhcos.mirror.openshift.com/art/storage/releases/rhcos-%s/%s/%s",
+                       ocpVersionF,
+                       rhcosVersion,
+                       coreosarch.CurrentRpmArch())
+               latestRhcosBuildMetaUrl = fmt.Sprintf("%s/meta.json", latestBaseUrl)
+               if err := getJson(latestRhcosBuildMetaUrl, &latestOcpRhcosBuild); err != nil {
+                       return "", err
+               }
        }

        latestRhcosQcow2 := latestOcpRhcosBuild.BuildArtifacts.Qemu.Path
```

(Untested)

Reply:
I tested it and added it verbatim.  Thanks @jlebon!